flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/FileSystemJobResultStoreTest.java

```diff
@@ -0,0 +1,253 @@
+
+    private FileSystemJobResultStore fileSystemJobResultStore;
+
+    @TempDir File temporaryFolder;
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    @BeforeEach
+    public void setupTest() throws IOException {
+        Path path = new Path(temporaryFolder.toURI());
+        fileSystemJobResultStore = new FileSystemJobResultStore(path.getFileSystem(), path, false);
+    }
+
+    /** Tests that adding a JobResult to the JobResultStore results in a Dirty JobResultEntry. */
+    @Test
+    public void testStoreDirtyJobResult() throws Exception {
+        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+        assertThat(fileSystemJobResultStore.hasJobResultEntry(DUMMY_JOB_RESULT_ENTRY.getJobId()))
+                .isTrue();
+
+        Set<JobResult> dirtyResults = fileSystemJobResultStore.getDirtyResults();
+        assertThat(dirtyResults.stream().map(JobResult::getJobId).collect(Collectors.toList()))
+                .containsExactlyInAnyOrder(DUMMY_JOB_RESULT_ENTRY.getJobId());
+        assertThat(getCleanResultIds()).isEmpty();
+
+        Path path = expectedDirtyPath(DUMMY_JOB_RESULT_ENTRY);
+        assertThat(new File(path.getPath())).exists().isFile().isNotEmpty();
+    }
+
+    @Test
+    public void testStoreDirtyJobResultTwice() throws IOException {
+        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+
+        assertThatThrownBy(() -> fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY))
+                .isInstanceOf(IllegalStateException.class);
+    }
+
+    @Test
+    public void testStoreDirtyJobResultAfterClean() throws IOException {
+        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+        fileSystemJobResultStore.markResultAsClean(DUMMY_JOB_RESULT_ENTRY.getJobId());
+
+        assertThatThrownBy(() -> fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY))
+                .isInstanceOf(IllegalStateException.class);
+    }
+
+    /**
+     * Tests that adding a JobResult to the JobResultsStore and then marking it as clean puts it
+     * into a clean state.
+     */
+    @Test
+    public void testCleanDirtyJobResult() throws Exception {
+        fileSystemJobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+        fileSystemJobResultStore.markResultAsClean(DUMMY_JOB_RESULT_ENTRY.getJobId());
+
+        assertThat(fileSystemJobResultStore.getDirtyResults()).isEmpty();
+        assertThat(getCleanResultIds())
+                .containsExactlyInAnyOrder(DUMMY_JOB_RESULT_ENTRY.getJobId());
+
+        Path path = expectedCleanPath(DUMMY_JOB_RESULT_ENTRY);
+        assertThat(new File(path.getPath())).exists().isFile().isNotEmpty();
+    }
+
+    /**
+     * Tests that adding a JobResult to the JobResultsStore and then marking it as clean puts it
+     * into a clean state.
+     */
+    @Test
+    public void testCleanDirtyJobResultTwice() throws Exception {
```

Comment:
This test tests the same thing the previous test method `testCleanDirtyJobResult` tests. Additionally, the test method name indicates that we want to test something else here.